powerstore/resource_host_group.go

```diff
@@ -0,0 +1,249 @@
+)
+
+// newHostGroupResource returns host group new resource instance
+func newHostGroupResource() resource.Resource {
+	return &resourceHostGroup{}
+}
+
+type resourceHostGroup struct {
+	client *client.Client
+}
+
+// Metadata defines resource interface Metadata method
+func (r *resourceHostGroup) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_hostgroup"
+}
+
+// Schema defines resource interface Schema method
+func (r *resourceHostGroup) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+
+		MarkdownDescription: "HostGroup resource",
+
+		Attributes: map[string]schema.Attribute{
+
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Unique identifier of the host group.",
+				MarkdownDescription: "Unique identifier of the host group.",
+			},
+
+			"name": schema.StringAttribute{
+				Required:            true,
+				Description:         "The host group name.",
+				MarkdownDescription: "The host group name.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"description": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				Description:         "An optional description for the host group.",
+				MarkdownDescription: "An optional description for the host group.",
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+
+			"host_ids": schema.SetAttribute{
+				ElementType:         types.StringType,
+				Required:            true,
+				Description:         "The list of hosts to include in the host group.",
+				MarkdownDescription: "The list of hosts to include in the host group.",
+				Validators: []validator.Set{
+					setvalidator.ValueStringsAre(
+						stringvalidator.LengthAtLeast(1),
+					),
+				},
+			},
+		},
+	}
+}
+
+// Configure - defines configuration for host group resource
+func (r *resourceHostGroup) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+// Create - method to create host group resource
+func (r *resourceHostGroup) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+
+	var plan models.HostGroup
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	hostGroupCreate, err := r.planToHostGroupParam(plan)
```

Comment:
Error will be always nil from the planToHostGroupParam method?

Reply:
yes, as there is no method call from planToHostGroupParam there is no error so, removed return parameter.